### PR TITLE
Return errors from the plugin functions to host 

### LINKED ISSUE #15: Errors returned from plugin are never surfaced to host
When a plugin function returns an error, it seems like that error never makes it's way back to the host.

Version: [f7d9444](https://github.com/knqyf263/go-plugin/commit/f7d94447ee0de525262d52bdc51d75e8fdc2fab6)
Reproducible Test: https://github.com/cchamplin/plugin-bug

Host
```go
	_, err = myPlugin.PassDataError(context.Background(), plugin.PassDataRequest{})
	if err == nil {
		t.Error("expected error")
		return
	}
```

Plugin
```go
func (p TestPlugin) PassDataError(ctx context.Context, request host.PassDataRequest) (result host.PassDataReply, err error) {
	return host.PassDataReply{}, fmt.Errorf("an error from the plugin")
}
```

Expected: Test to pass
Actual:
```text
=== RUN   TestPlugin_Error
    plugin_test.go:184: expected error
--- FAIL: TestPlugin_Error (0.10s)

FAIL
```

### PR DESCRIPTION
**Pull request by dmvolod**
*Issue #, if available:*
Fixes #15 

*Description of changes:*
This fix sends error message occurrence sign within high bit of the offset/len field.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/fields/fields_test.go [hunk 1]
```diff
@@ -77,3 +72,23 @@ func TestFields(t *testing.T) {
...
 	assert.Equal(t, want, got)
 }
+
+func TestErrorResponse(t *testing.T) {
+	ctx := context.Background()
+	plugin := loadPlugin(ctx, t)
+
+	_, err := plugin.TestError(ctx, emptypb.Empty{})
+	require.Error(t, err)
+	require.Equal(t, err.Error(), "error from plugin")
```
Comment (mathetake) [inline tests/fields/fields_test.go:82]: can we add an additional case to ensure that empty error (errors.New("")) is also correctly handled?

File: gen/host.go [hunk 2]
```diff
@@ -325,23 +325,36 @@ func genPluginMethod(g *protogen.GeneratedFile, f *fileInfo, method *protogen.Me
...
 	g.P(errorHandling)
-	g.P(`
+	g.P(fmt.Sprintf(`
 			// Note: This pointer is still owned by TinyGo, so don't try to free it!
 			resPtr := uint32(ptrSize[0] >> 32)
 			resSize := uint32(ptrSize[0])
+			var isErrResponse bool
+			if (resSize & %s) > 0 {
+			isErrResponse = true
+				resSize &^= %s
```
Comment (mathetake) [inline gen/host.go:335]:
```suggestion
			if (resSize & %s) > 0 {
				isErrResponse = true
				resSize &^= %s
```

**Comment by mathetake** [comment]
left some comments, but looking good!

File: gen/host.go (hunk 2 above)
Comment (dmvolod) [reply gen/host.go]: Thanks for feedback. Fixed.

File: tests/fields/fields_test.go (hunk 1 above)
Comment (dmvolod) [reply tests/fields/fields_test.go]: Yes, sure. Additional test added.

**Review by mathetake — APPROVED** [review]
awesome!

**Review by codefromthecrypt — APPROVED** [review]
good job!

**Pull request merged**